Controller/Api/ApiController.php

```diff
@@ -80,4 +98,194 @@ public function handlerAction(
 
         return $this->handleView($view);
     }
+
+    /**
+     * @param $campaignId
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function listCampaignSourcesAction($campaignId)
+    {
+        $table_alias    = $this->model->getRepository()->getTableAlias();
+        $campaignFilter = [
+            'column' => $table_alias.'.campaign_settings',
+            'expr'   => 'like', //like
+            'value'  => '%"campaignId": "'.$campaignId.'"%',
```

Comment:
This is dangerous without typecasting the campaignID to an int.

Secondly, is this even necessary if you are intending to implement campaignList?

Reply:
type casted via intval

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/8c15569e-920d-40ca-a005-52d026c9bde5-c1dd3038-7698-4e2e-8c77-ea3eee7b235f-v1.png?transparent=1&palette=1&width=246)